doc/graph/sdpa_with_compressed_kv.md

```diff
@@ -0,0 +1,124 @@
+SDPA with Compressed Key and Value {#dev_guide_graph_compressed_sdpa}
+===========================================================
+
+## Overview
+
+The KV Cache mechanism was developed to improve the efficiency of models by
+reducing computational redundancy when processing long sequences of data. It
+stores previously computed hidden states of Key and Value, to enable faster
+retrieval and speed up inference processes. However, with the growing popularity
+of KV Cache, the memory required for caching has become a significant bottleneck
+affecting model performance.
+
+To address this, Scaled Dot-Product Attention (SDPA)[1] with compressed Keys and
+Values is implemented to minimize the memory footprint during generative
+inference in large language models. Specifically, Key and Value tensors are
+stored using lower precision data types like int4 and int8 to reduce memory
+usage, and are subsequently de-quantized to wider floating point types such as
+f16 and bf16 for computation.
+
+It's worth noting that grouped quantization is required to improve model
+accurarcy, especially for int4 data types. In this case, group size is needed
+as an attribute for quantization, which indicates the number of elements that
+share the same scaling factor and zero points in each quantization group.
```

Comment:
```suggestion
share the same scaling factor and zero-points in each quantization group.
```
please use a `-` between zero and point here and all other occurences, as without a `-` it has a different meaning. :)